artiq/gateware/eem_7series.py

```diff
@@ -19,6 +19,21 @@ def peripheral_dio(module, peripheral, **kwargs):
         edge_counter_cls=edge_counter_cls, **kwargs)
 
 
+def peripheral_dio_spi(module, peripheral, **kwargs):
+    ttl_classes = {
+        "input": ttl_serdes_7series.InOut_8X,
+        "output": ttl_serdes_7series.Output_8X
+    }
+    if len(peripheral["ports"]) != 1:
```

Comment:
Is this checked in the schema?

Reply:
*I think this is checked in the schema. so is the input never validated against schema?

Reply:
This is also defined in the json schema. If I understand correctly, the json files are checked against the schema, so this check is a bit redundant. Anyhow, code for other EEM devices also checks the number of ports so I also just replicated that behavior.